logstreams/src/test/java/io/zeebe/logstreams/impl/log/LogStreamReaderTest.java

```diff
@@ -29,7 +30,6 @@
 public final class LogStreamReaderTest {
```

Comment:
Hm, I assumed there were, but I imagine there aren't because, the reader doesn't know much details about the writer, i.e. what is the middle of the batch, the full batch, etc.

I'll add one that we check all events are ordered though, as that's expected from the reader (otherwise everything breaks).

Reply:
In `shouldIterateOverManyEvents` it is multiple single records, right? LogStorageReader returns a batch of Records. It is the responsibility of LogStreamReader to split them into multiple records.So I think, we should add a test for it.

Reply:
It is not a new behavior introduced in this PR. So we can also address it in a follow up issue.

Reply:
I've switched the write rule to use the batch writer, as this is our common use case. So both tests now also test with batches.